presto-main/src/main/java/com/facebook/presto/server/StatementResource.java

```diff
@@ -450,6 +450,14 @@ else if (!queryInfo.getOutputStage().isPresent()) {
                 }
             }
 
+            List<QueryError> warnings;
+            if (queryInfo.getState().isDone()) {
+                warnings = queryInfo.getWarnings();
```

Comment:
Don't know the context here, but why can't we get warnings of yet to be finished a query? Especially that warnings are now generated by analysis (AFAIU), which is quite early.